### PR TITLE
Add Darwin notification listener to Test App for FLW testing

### PR DESCRIPTION
**Pull request by mipetriu**
## Proposed changes

As part of FLW signout, the broker will send a system-wide broadcast to notify any listeners that the signout occurred. This PR adds a test listener to the MSAL test app to log when this broadcast reaches the app.

## Type of change

- [ ] Feature work
- [ ] Bug fix
- [ ] Documentation
- [ ] Engineering change
- [x] Test
- [ ] Logging/Telemetry

## Risk

- [ ] High – Errors could cause MAJOR regression of many scenarios. (Example: new large features or high level infrastructure changes)
- [x] Medium – Errors could cause regression of 1 or more scenarios. (Example: somewhat complex bug fixes, small new features)
- [ ] Small – No issues are expected. (Example: Very small bug fixes, string changes, or configuration settings changes)

### CODE REVIEW

File: MSAL/test/app/ios/MSALTestAppAcquireTokenViewController.m [hunk 1]
```diff
@@ -59,6 +59,9 @@
 
 static NSString *const kDeviceIdClaimsValue = @"{\"access_token\":{\"deviceid\":{\"essential\":true}}}";
 
+static NSString *const kGlobalSignoutDarwinNotificationKey = @"FLWSignoutOccured";
```
Comment (oldalton) [inline MSAL/test/app/ios/MSALTestAppAcquireTokenViewController.m:62]: Should we define those in MSAL API somewhere?

**Review by hieunguyenmsft — APPROVED** [review]
Just curious, if the test app in foreground still receive notification?

File: MSAL/test/app/ios/MSALTestAppAcquireTokenViewController.m (hunk 1 above)
Comment (mipetriu) [reply MSAL/test/app/ios/MSALTestAppAcquireTokenViewController.m:62]: At this point, it seems that the Darwin notification solution is a targeted solution to only 1 partner app. If it seems that we will need to use it in other apps, I agree that it should be defined more generally in the MSAL API.

**Comment by mipetriu** [comment]
@hieunguyenmsft - The app needs to be in a non-suspended state to be able to receive the notification. That means that the app can be active, or running in the background doing some other task that's allowed by Apple. Just listening to this notification is not a reason enough to receive the background permission from Apple, so this solution is pretty limited to apps that are doing something else in the background, but want to listen to the notification alongside that.

**Comment by hieunguyenmsft** [comment]
> @hieunguyenmsft - The app needs to be in a non-suspended state to be able to receive the notification. That means that the app can be active, or running in the background doing some other task that's allowed by Apple. Just listening to this notification is not a reason enough to receive the background permission from Apple, so this solution is pretty limited to apps that are doing something else in the background, but want to listen to the notification alongside that.

I see, but AFAIK the test app doesn't perform the background task, so are you adding new background for it to run? Or (just guessing) you open the webview and bring the testapp to foreground?

File: MSAL/test/app/ios/MSALTestAppAcquireTokenViewController.m [hunk 2]
```diff
@@ -763,4 +775,18 @@ - (void)hideCustomeWebViewIfNeed
...
+    self.resultTextView.text = @"Darwin notification received from the broker SDK indicating a global signout has occurred.";
+}
+
+void globalSignoutCallback(__unused CFNotificationCenterRef center,
+                           __unused void * observer,
+                           __unused CFStringRef name,
+                           __unused void const * object,
+                           __unused CFDictionaryRef userInfo)
+{
+    [[NSNotificationCenter defaultCenter] postNotificationName:kDarwinNotificationReceivedKey object:nil];
```
Comment (oldalton) [inline MSAL/test/app/ios/MSALTestAppAcquireTokenViewController.m:789]: Any reason why a separate NSNotificationCenter callback is required?

Comment (mipetriu) [reply MSAL/test/app/ios/MSALTestAppAcquireTokenViewController.m:789]: The globalSignoutCallback is a C function that has a different set of rules for how it recognizes the other functions within the class. I took inspiration from one of the ways Teams uses the Darwin callback in their code for a quick way to get it working.

Comment (antrix1989) [reply MSAL/test/app/ios/MSALTestAppAcquireTokenViewController.m:789]: can we prefix this funciton the same way as we do with other constants/global functions?

**Comment by antrix1989** [comment]
Where is IdenityCore pr? Don't see it there...

File: MSAL/test/app/ios/MSALTestAppAcquireTokenViewController.m (hunk 1 above)
Comment (mipetriu) [reply MSAL/test/app/ios/MSALTestAppAcquireTokenViewController.m]: After further discussions, we will need to expand this for other apps, so I created an MSID constant

**Pull request merged**